sdv/constraints/base.py

```diff
@@ -250,44 +169,16 @@ def _validate_data_on_constraint(self, table_data):
 
                 raise ConstraintsNotMetError(err_msg)
 
-    def _validate_constraint_columns(self, table_data):
-        """Validate the columns in ``table_data``.
-
-        If ``fit_columns_model`` is False and any columns in ``constraint_columns``
-        are not present in ``table_data``, this method will raise a
-        ``MissingConstraintColumnError``. Otherwise it will return the ``table_data``
-        unchanged. If ``fit_columns_model`` is True, then this method will sample
-        any missing ``constraint_columns`` from its model conditioned on the
-        ``constraint_columns`` that ``table_data`` does contain. If ``table_data``
-        doesn't contain any of the ``constraint_columns`` then a
-        ``MissingConstraintColumnError`` will be raised.
+    def check_missing_columns(self, table_data):
+        """Check ``table_data`` for missing columns.
 
         Args:
             table_data (pandas.DataFrame):
                 Table data.
         """
         missing_columns = [col for col in self.constraint_columns if col not in table_data.columns]
         if missing_columns:
-            if not self._columns_model:
-                warning_message = (
-                    'When `fit_columns_model` is False and we are conditioning on a subset '
-                    'of the constraint columns, conditional sampling uses reject sampling '
-                    'which can be slow. Changing `fit_columns_model` to True can improve '
-                    'the performance.'
-                )
-                warnings.warn(warning_message, UserWarning)
-
-            all_columns_missing = len(missing_columns) == len(self.constraint_columns)
```

Comment:
Why did you remove this line?

Reply:
Since we don't have the `columns_model` in here anymore, and we raise the error whenever a column is missing (doesn't really matter if it's one or all).